pkg/vm/run_context.go

```diff
@@ -9,3 +13,61 @@ type RunContext struct {
 	Ap memory.Relocatable
 	Fp memory.Relocatable
 }
+
+func (run_context RunContext) ComputeDstAddr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.DstReg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.OffOp0 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.OffDst))))
+	} else {
+		return base_addr.AddUint(uint(instruction.OffDst))
+	}
+
+}
+
+func (run_context RunContext) ComputeOp0Addr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.Op0Reg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.OffOp1 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.OffOp0))))
+	} else {
+		return base_addr.AddUint(uint(instruction.OffOp0))
+	}
+}
+
+func (run_context RunContext) ComputeOp1Addr(instruction Instruction, op0 memory.MaybeRelocatable) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+
+	switch instruction.Op1Addr {
+	case Op1SrcFP:
+		base_addr = run_context.Fp
+	case Op1SrcAP:
+		base_addr = run_context.Ap
+	case Op1SrcImm:
+		if instruction.OffOp1 == 1 {
+			base_addr = run_context.Pc
+		} else {
+			base_addr = memory.NewRelocatable(-1, 0)
+			return base_addr, &VirtualMachineError{Msg: "UnknownOp0"}
```

Comment:
The convention in go is to return the zero-value in these cases, so we should probably return Relocatable{} (aka (0,0))